pkg/k8s/secrets/providers/k8s/provider.go

```diff
@@ -36,5 +36,6 @@ func (p *provider) GetSecret(name string) (*v1.Secret, error) {
 	if !secret.ObjectMeta.DeletionTimestamp.IsZero() {
 		return nil, nil
 	}
-	return providers.StripMetadata(secret), nil
+	providers.StripMetadata(secret)
```

Comment:
Are we sure we don't need to deep-copy the secret here as well?

Reply:
The underlying client provides us with new objects per get request. Thererfore copying is not necessary here.